src/Perlang.Interpreter/NameResolution/NameResolver.cs

```diff
@@ -424,7 +425,7 @@ public VoidObject VisitVarStmt(Stmt.Var stmt)
                 Resolve(stmt.Initializer);
             }
 
-            Define(stmt.Name, stmt.Initializer?.TypeReference ?? new TypeReference(stmt.Name));
+            Define(stmt.Name, stmt.TypeReference ?? new TypeReference(stmt.Name));
```

Comment:
Oops, this is a bug fix that should go in but not as early as in this PR. Will remove (and perhaps remove other tests which fail because of it) before merge.

Reply:
Fixed, resolving.